### PR TITLE
w3m 0.5.3-38

### PR DESCRIPTION
**Pull request by fxcoudert**
Debian has become upstream, really.

### CODE REVIEW

**Comment by retokromer** [comment]
To my knowledge, it’s a MIT license.

File: Formula/w3m.rb [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
-      sha256 "625f5b0cb71bf29b67ad3bb9c316420922877473a6e94e6c7bcc337cb22ce1eb"
+      url "https://deb.debian.org/debian/pool/main/w/w3m/w3m_0.5.3-38.debian.tar.xz"
+      sha256 "227dd8d27946f21186d74ac6b7bcf148c37d97066c7ccded16495d9e22520792"
       apply "patches/010_upstream.patch",
             "patches/020_debian.patch"
     end
   end
 
   livecheck do
     url :stable
```
Comment (nandahkrishna) [inline Formula/w3m.rb:22]:
```suggestion
    url "https://deb.debian.org/debian/pool/main/w/w3m/"
    regex(%r{href=.*?w3m[._-]v?(\d+(?:\.\d+)+)\.orig\.t}i)
```

File: Formula/w3m.rb [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
-    regex(%r{url=.*?/w3m[._-]v?(\d+(?:\.\d+)+)\.t}i)
+    regex(%r{url=.*?/w3m[._-]v?(\d+(?:\.\d+)+)\.orig.t}i)
```
Comment (nandahkrishna) [inline Formula/w3m.rb:23]: I'm not too familiar with the suggestions thing, but this should be:
```
  url "https://deb.debian.org/debian/pool/main/w/w3m/"
  regex(%r{href=.*?w3m[._-]v?(\d+(?:\.\d+)+)\.orig\.t}i)
```
This ensures `0.5.3` is still reported as the version found by livecheck.

**Comment by mitchblank** [comment]
@fxcoudert -- you might want to try rebottling `nmh` as well.  It had `w3m` as a dependency but build+test works fine for me locally.

**Comment by fxcoudert** [comment]
If the linkage test and the tests worked fine, why would we rebottle nmh?

**Comment by mitchblank** [comment]
> If the linkage test and the tests worked fine, why would we rebottle nmh?

Just so we can get a big_sur native bottle for nmh so we can get it off of the #65000 list.  It didn't happen duing the mass-rebottling, but that was possibly due to the w3m dependency, since it builds from source fine for me.

**Comment by fxcoudert** [comment]
I can dispatch a Big Sur bottle build without revision bumping, just ping me when you see a formula that should build.

**Comment by mitchblank** [comment]
Is there a particular place that is best to ping you at?  My rebottling wishlist is getting long.

**Comment by fxcoudert** [comment]
Any issue, here is fine :)

**Comment by mitchblank** [comment]
Here are some entries from your #65000 list that either have had dependencies resolved OR I couldn't find out what had failed when I ran them manually.  Some of them I've pinged people on other PRs as noted by the links:
```
2621 mingw-w64 -- compile/test worked fine for me manually; possibly just needs a rebottle now that more dependencies are done
1450 dpkg -- should be OK now that "po4a" dependency is fixed https://github.com/Homebrew/homebrew-core/pull/67336#issuecomment-749838688
176 bitcoin -- maybe will bottle now https://github.com/Homebrew/homebrew-core/pull/66852#issuecomment-744409428
147 bbftp-client -- built and tested fine for me manually
105 cabocha -- crf++ dependency fixed; probably just needs a re-bottle https://github.com/Homebrew/homebrew-core/pull/67265#issuecomment-748594850
97 libofx -- now may be ready for a rebottle; https://github.com/Homebrew/homebrew-core/pull/66986#issuecomment-748560325
72 dict -- maybe will rebottle now that libmaa depdency is fixed https://github.com/Homebrew/homebrew-core/pull/67266#issuecomment-748595185
58 launchdns -- compiled and tested fine for me, not sure what CI's problem was
26 lemon -- build/test manually worked for me
20 amap -- build/test worked ok for me manually
15 zpaq -- fix is merged, but needs a rebottling job kicked off
14 cvs-fast-export -- builds/installs fine.  probably failed to bottle because of "cvs" dependency which is already fixed
13 oq -- dependency sorted, maybe can be bottled? https://github.com/Homebrew/homebrew-core/pull/66796#issuecomment-748944514
8 get-flash-videos -- install+test worked fine for me, not sure what the bottling issue is
7 crystal-icr -- dependency sorted, maybe can be bottled? https://github.com/Homebrew/homebrew-core/pull/66796#issuecomment-748944514
5 zyre -- was broken due to czmq dependency, but may need more... or maybe test just doesn't like Little Snitch
1 reprepro -- maybe will bottle now https://github.com/Homebrew/homebrew-core/pull/66852#issuecomment-744409428

=== If mingw-64 succeeds above then do:

111 makensis -- depends_on "mingw-w64"

=== If libofx succeeds above then do:

113 homebank -- depends_on "libofx"
```
I don't know if all of them will bottle for Big Sur now but I bet some of them will.

**Comment by fxcoudert** [comment]
@mitchblank I've queued bottles for all of those listed above, the failures should be linked here below

**Comment by mitchblank** [comment]
Well it looks like we got 12 successes and 6 failures.
```
bbftp-client.rb:    sha256 "74cbb945c2451974029b924f8ee6dc121b15c4dd301873560afd39d37f930686" => :big_sur
bitcoin.rb:    sha256 "cc325b9d083a2ecb7816ffc61cb4de7ac78e45cb56ecadce26132283595b83a8" => :big_sur
cabocha.rb:    sha256 "bd52ed1080db697c45768449954fcbc1ead6b41bd6a2481b843a1d7177155b34" => :big_sur
crystal-icr.rb:    sha256 "fcddd33608c0eccf38cd3f3e7bc299cfac6ca2b7020e7d3171eead79c932b2db" => :big_sur
cvs-fast-export.rb:    sha256 "af7e3b2fa6fd3c2e4095364bbf13af5e1aa9c00f7ff066f95f2b8485ce4b22b5" => :big_sur
dict.rb:    sha256 "e2f72aa23fa4294603f7c85da2c0748e97c36360e18845c929fc0de691ef8952" => :big_sur
dpkg.rb:    sha256 "4f28e3cf9f5efab34cfeca360751e5369552e9a5122e2b76b6375e0c7425999f" => :big_sur
libofx.rb:    sha256 "b7cfdbbdf131872ff2c0cb26c4e14ee2c346fa8e16c8e0293be58f4c60d43e2e" => :big_sur
makensis.rb:    sha256 "aa8a346937316765bf9ffe7d532b08212fab4ae697aad7e23185baeabe280249" => :big_sur
reprepro.rb:    sha256 "3064f238486aad600cf095168f03513679159bbbddbf50646d64be1bac1a8d36" => :big_sur
zpaq.rb:    sha256 "05c7f10eb18f64d77057a80ce7580bdf8dffd9d33e743259dae3984803dc9dd9" => :big_sur
zyre.rb:    sha256 "092d6682e5c0ca01066bdf991599e8104d382a80f5793ef1c9f57dcdfb5689eb" => :big_sur
```
@fxcoudert -- since `libofx` build OK could you do `homebank` now?  It builds/tests fine for me locally now.

**Comment by mitchblank** [comment]
> ❌ @fxcoudert bottle request for launchdns [failed](https://github.com/Homebrew/homebrew-core/actions/runs/440307458).
This failed immediately due to:
```
==> brew audit launchdns --online --git --skip-style
==> FAILED
launchdns:
  * 'revision 1' should be removed
Error: 1 problem in 1 formula detected
```

Sure enough, when this was updated from 1.0.3 -> 1.0.4 the `revision` line was accidentally left on: 91c7119b03c26a837e62cf8f86ef84f0bd3c0f76

However I don't know how to resolve the problem *now*.  If I remove the `revision 1` after we already have `1.0.4_1` bottles won't that look like a downgrade?

**Comment by mitchblank** [comment]
> ❌ @fxcoudert bottle request for lemon [failed](https://github.com/Homebrew/homebrew-core/actions/runs/440307558).
Opened #67584 to fix download url (although the formula may have other long-term issues, as described)

**Comment by mitchblank** [comment]
> ❌ @fxcoudert bottle request for amap [failed](https://github.com/Homebrew/homebrew-core/actions/runs/440307659).

This failed to bottle because of a /usr/local/Cellar path ending up in a binary which I guess is a no-no somehow?
```
==> Bottling amap--5.4_3.big_sur.bottle.tar.gz...
/usr/bin/sudo --non-interactive /usr/sbin/purge
tar cf /Users/brew/actions-runner/_work/homebrew-core/homebrew-core/bottles/amap--5.4_3.big_sur.bottle.tar amap/5.4_3
/usr/bin/sudo --non-interactive /usr/sbin/purge
gzip -f amap-bottle.tar
/usr/bin/sudo --non-interactive /usr/sbin/purge
Warning: String '/usr/local/Cellar' still exists in these files:
/usr/local/Cellar/amap/5.4_3/bin/amap
 --> match '/usr/local/Cellar/amap/5.4_3' at offset 0x124de

Error: --keep-old was passed but there are changes in:
cellar: old: :any, new: "/usr/local/Cellar"
```
I don't fully understand what it is complaining about though.

**Comment by mitchblank** [comment]
> ❌ @fxcoudert bottle request for get-flash-videos [failed](https://github.com/Homebrew/homebrew-core/actions/runs/440308182).

This one failed because the audit step no longer allows "prerelease" downloads from github:
```
==> brew audit get-flash-videos --online --git --skip-style
==> FAILED
get-flash-videos:
  * 1.25.99.03 is a GitHub pre-release.
Error: 1 problem in 1 formula detected
```
I don't know the exact procedure for fixing this.  I opened #67586 to add an audit_exception for this, we'll see if that gets accepted.

**Comment by mitchblank** [comment]
> ❌ @fxcoudert bottle request for oq [failed](https://github.com/Homebrew/homebrew-core/actions/runs/440308101).

Bottling failed due to a missing dependency on `pcre`.  Hopefully #67587 will fix; we'll see when its CI completes.

**Comment by mitchblank** [comment]
> ❌ @fxcoudert bottle request for mingw-w64 [failed](https://github.com/Homebrew/homebrew-core/actions/runs/440308603).

This is an issue with some shim paths being left behind; I'll take a look at it:
```
==> brew audit mingw-w64 --online --git --skip-style
==> FAILED
mingw-w64:
  * Files were found with references to the Homebrew shims directory.
    The offending files are:
      toolchain-i686/libexec/gcc/i686-w64-mingw32/10.2.0/install-tools/fixincl
      toolchain-x86_64/libexec/gcc/x86_64-w64-mingw32/10.2.0/install-tools/fixincl
Error: 1 problem in 1 formula detected
```

**Comment by gromgit** [comment]
> > x @fxcoudert bottle request for amap [failed](https://github.com/Homebrew/homebrew-core/actions/runs/440307659).
> 
> This failed to bottle because of a /usr/local/Cellar path ending up in a binary which I guess is a no-no somehow?

It looks more like:

> ```
> Warning: String '/usr/local/Cellar' still exists in these files:
> /usr/local/Cellar/amap/5.4_3/bin/amap
>  --> match '/usr/local/Cellar/amap/5.4_3' at offset 0x124de
> 
> Error: --keep-old was passed but there are changes in:
> cellar: old: :any, new: "/usr/local/Cellar"
> ```

The formula says `amap` bottles should be relocatable, but the build check says "hang on, there's a cellar ref in here".

**Comment by mitchblank** [comment]
> The formula says `amap` bottles should be relocatable,

So just needs `cellar :any` removed?

**Comment by gromgit** [comment]
> So just needs `cellar :any` removed?

Better to investigate further, since:
```sh
$ strings - /usr/local/Cellar/amap/5.4_3/bin/amap | grep Cellar
# no output
```
the existing Catalina bottle at least doesn't embed Cellar refs, so there's no obvious reason why this is happening.

**Comment by gromgit** [comment]
Huh, turns out the Big Sur build is actually correct: previous builds are technically b0rken, and `amap` is technically _not_ relocatable.

`amap` embeds `$HOMEBREW_PREFIX` as a root path to find its data files, but a `sed` bug in its configure script:
```
PREFIX=`echo "$1"|grep prefix=|sed 's/.*prefix=//i'`
```
actually causes pre-Big Sur builds to incorrectly embed its default `/usr/local` instead, because the older macOS `sed` doesn't recognize the `i` (case-insensitive) modifier. That's probably the underlying reason for the `# --prefix doesn't work as we want it to so install manually` section in the formula.

Also, the current test block doesn't trigger on the incorrect path, because it's effectively a version check. That too needs to be fixed, though very carefully (it's a port-scanner, so basically a loaded Avenger cannon).

So I can PR the necessary fixes, but will need a command decision: Should we deprecate this instead? As the formula notes:
```
# Last release was 2011 & there's nowhere supported to report this.
```
and usage stats aren't impressive:
```
install-on-request: 23 (30 days), 79 (90 days), 303 (365 days)
```

**Comment by gromgit** [comment]
After a much-needed nap, my previous comment looks a little rambling, so here's the executive summary:
1. `amap` cannot be made relocatable without significant code changes
2. existing bottles are incorrectly marked as relocatable, and will fail to run under non-standard prefixes
3. the `amap` code itself indicates that it's no longer maintained
4. the upstream URL is actually a "code boneyard"
5. it's not much used

Therefore, I recommend disabling it immediately, as the last code change was in 2011, and the latest reasonable deprecation date (the date it was added to the "boneyard") is 4 Jun 2018, over two years ago.

@fxcoudert, do you concur?

**Comment by SMillerDev** [comment]
I think we can disable it starting 2 years ago.

**Comment by gromgit** [comment]
OK, will do the necessary.

**Comment by mitchblank** [comment]
> ❌ @fxcoudert bottle request for get-flash-videos [failed](https://github.com/Homebrew/homebrew-core/actions/runs/442178988).

Arg!!!!
```
==> FAILED
get-flash-videos:
  * 1.25.99.03 is not a GitHub pre-release but 'get-flash-videos' is in the GitHub prerelease allowlist.
Error: 1 problem in 1 formula detected
```

So if we don't include it on the whitelist, audit fails due to it being a prerelease.

If we *do* include it, audit fails because suddenly it's *not* a prerelease.

How do we make this happy?  cc @Rylan12

**Comment by Rylan12** [comment]
That's... odd. Seems like an issue with our prerelease detection (or I made a logic mistake somewhere). It's definitely listed as a prerelease: https://github.com/monsieurvideo/get-flash-videos/releases/tag/1.25.99.03

Let me look

**Comment by Rylan12** [comment]
@mitchblank this was a bug I introduced. I've opened a PR to fix it: https://github.com/Homebrew/brew/pull/10138. That PR is set to automerge once CI passes. Should be fixed then.

**Comment by Rylan12** [comment]
@mitchblank sorry, forgot to ping when this was merged. Do you need a workflow to be rerun? Let me know if you do.

**Comment by mitchblank** [comment]
> @mitchblank sorry, forgot to ping when this was merged. Do you need a workflow to be rerun? Let me know if you do.

If you could kick off another rebottle job of `get-flash-videos` that would be great.  I'm just trying to figure out why that formula hasn't been bottling correctly lately which is why I was messing with the prerelease whitelist.

**Comment by Rylan12** [comment]
@mitchblank here you go: https://github.com/Homebrew/homebrew-core/actions/runs/443355277

**Comment by mitchblank** [comment]
@fxcoudert -- another rebottling request, please: `dfc`

The project's website was dead which may have been why rebottling failed before.  I contacted the maintainer and the have fixed it.

**Comment by fxcoudert** [comment]
`dfc` bottling dispatched

**Comment by mitchblank** [comment]
> `dfc` bottling dispatched

Thanks, it worked:
```
    sha256 "45b7f955cfc8d2acbb1b949abbb4797a92fe63f7820508b1f1d44d4da601479f" => :big_sur
```

**Comment by mitchblank** [comment]
@fxcoudert -- another set of rebottling requests if you have a moment.  All of these build+tested fine for me manualy
* `libcddb`
* `zorba`
* `traildb`
* `hqx`
* `imagejs`     (audit might have failed due to the homepage being wrong.  fixed in #67752 but change didn't trigger a rebottle)
* `augustus`   (I originally tried just upgrading formula to get a rebottle but hit issues: https://github.com/Homebrew/discussions/discussions/368 )

**Comment by mitchblank** [comment]
Well, 3 of 6 in that batch:
```
libcddb.rb:    sha256 "14faa4ce2a853bbe09dbaf20e78b820728690b4e27b7cb1a7bd956a7b6a9ee85" => :big_sur
hqx.rb:    sha256 "00a56bf946edffc932666fa52e0962a0a1bd2f10c23dad1b9751fe74d9ae04fc" => :big_sur
imagejs.rb:    sha256 "7b9812855ddca5e1553eed1e613a840642d01fbd35e5c5558610a7a96f7142a6" => :big_sur
```

**Comment by mitchblank** [comment]
> ❌ @fxcoudert bottle request for zorba [failed](https://github.com/Homebrew/homebrew-core/actions/runs/445581022).

OK looks like it needs bison at build time.  (It has a runtime dependency on flex already.. not sure why that isn't buildtime only, I'll have to look.

However the final failure was java related:
```
cd /tmp/zorba-20201226-24467-11nu0dh/zorba-3.1/test/rbkt/modules/java && /usr/bin/javac -classpath :/tmp/zorba-20201226-24467-11nu0dh/zorba-3.1/test/rbkt/modules/java:/tmp/zorba-20201226-24467-11nu0dh/zorba-3.1/build/test/rbkt/modules/java -d /tmp/zorba-20201226-24467-11nu0dh/zorba-3.1/build/test/rbkt/modules/java/CMakeFiles/JavaTest2.dir @/tmp/zorba-20201226-24467-11nu0dh/zorba-3.1/build/test/rbkt/modules/java/CMakeFiles/JavaTest2.dir/java_sources
[...]
The operation couldn’t be completed. Unable to locate a Java Runtime.
Please visit http://www.java.com for information on installing Java.
```

**Comment by mitchblank** [comment]
> ❌ @fxcoudert bottle request for traildb [failed](https://github.com/Homebrew/homebrew-core/actions/runs/445581055).

Bottle claims to be relocatable but seems not to be:
```
Warning: String '/usr/local/Cellar' still exists in these files:
/usr/local/Cellar/traildb/0.6/lib/libtraildb.a
 --> match '/usr/local/Cellar/libarchive/3.4.3/include' at offset 0x32ba3
```

**Comment by mitchblank** [comment]
> ❌ @fxcoudert bottle request for augustus [failed](https://github.com/Homebrew/homebrew-core/actions/runs/445581377).

`brew test` failed... not clear to me why, it worked when I built it manually:
```
==> /usr/local/Cellar/augustus/3.3.3_1/bin/augustus --species=human test.fasta

Error: augustus: failed
An exception occurred within a child process:
  Test::Unit::AssertionFailedError: <0> expected but was
<nil>.
```

**Comment by mitchblank** [comment]
> > ❌ @fxcoudert bottle request for augustus [failed](https://github.com/Homebrew/homebrew-core/actions/runs/445581377).
> 
> `brew test` failed... not clear to me why, it worked when I built it manually:

To make it even more annoying, this *did* bottle happily on arm64!

Looking at remaining items on the #65000 intel-big-sur list which *do* have `:arm64_big_sur` bottles:
```
bit.rb:    sha256 "616bf52a2d9e825320d9aab1af3603ee79e43f6d1455abd09576691049b70f0e" => :arm64_big_sur
nyancat.rb:    sha256 "e15caf1bc3a60334cb7e38466e50005cfbb15dd5e143abc876983fcb3d3a41d7" => :arm64_big_sur
airshare.rb:    sha256 "39e35bc89e09e6c7b106712658d222bb85640adb528d7dac368b9a728be17438" => :arm64_big_sur
nauty.rb:    sha256 "6d7980d67e772addbc5220e4536a1193580253ae2a52fbc7182d12eb73573c6b" => :arm64_big_sur
bluepill.rb:    sha256 "d7962bc6a7cebe51127a8195f15562298749880004a29ecc5f7ce688ff59a8d7" => :arm64_big_sur
task-spooler.rb:    sha256 "fd413989afe1cd32a45c02ed58b79dcab5c0242289a1b7ebf028228b4359aa39" => :arm64_big_sur
networkit.rb:    sha256 "23f70cc4fa3dc267391415b12f871ef39e27cca750e2e3a638b9075075ef4314" => :arm64_big_sur
augustus.rb:    sha256 "03665ee18df08813482e9ad6b91ff670ea332f828f6a0873ceef68dc0d574048" => :arm64_big_sur
dnsprobe.rb:    sha256 "740bcb0256a1da0e2d89d8a61e82e30eaecef7ac9766ab69b48cc2d5b678858e" => :arm64_big_sur
```
There are two other ones (`nyancat`, `airshare`) that I'd previously looked at and found no problems with a manual build, yet the CI runs had issues.

So either the intel Big Sur CI had issues which are now fixed, or there is something about the environment that is stricter than the arm64 CI... possibly around things like pty and network use.

It might be interesting to try bottling the other formula that have been failing on Big Sur CI for unclear reasons and see if arm64 accepts them (`curlpp` and `dungeon` come to mind...)

**Comment by mitchblank** [comment]
@fxcoudert -- could I bug you for some more rebottle jobs? 12 in particular.

These are formulae that could use an 11-Intel rebottle job kicked off:

* `thrift@0.9` -- failed before due to github prerelease status but exception now added.
* `cmt` -- should just need a rebottle; github tarballs had been previously marked as prereleases but that is now fixed upstream
* `couchdb` -- `erlang@22` dependency is available now on Big Sur
* `wrangler` -- `erlang@22` dependency is available now on Big Sur
* `scriptcs` -- `mono` dependency is available now on Big Sur
* `dar` -- `upx` build dependency is available now on Big Sur (Note: build won't succeed on ARM yet due to upx's current limitations)
* `pam_yubico` -- `ykclient` depdency is available now on Big Sur
* `pony-stable` -- `ponyc` dependency is available now on Big Sur (Intel, at least)
* `nauty` -- arm64 bottled OK and intel builds manually for me
* `libevhtp` -- homepage url fixed, should rebottle ok now  (arm64 already bottled OK)
* `minbif` -- PR to fix homepage already merged just needs a rebottle
* `bit` -- arm64 bottled OK and it works locally on intel for me

Separately, there are also a bunch of formula that still seem to be stuck on 11-Intel just due to CI sandboxing issues on that platform alone (note: 11-ARM generally seems to pass fine!)  I don't really know if there's anything at this point I can do to help diagnose these:
* `airshare`
* `augustus`
* `dnsprobe`
* `ncp`
* `nyancat`

**Comment by fxcoudert** [comment]
@mitchblank I've triggered those bottling jobs, failures should be posted here

**Comment by mitchblank** [comment]
> I've triggered those bottling jobs, failures should be posted here

Great; of the first 12, 10 of them now are bottled.  Exceptions:
* `nauty` failed to build because the website is down.  I emailed the project maintainer who is now looking into it.
* `couchdb` joins the list of formula which build fine but have CI issues on Big Sur (probably due to sandbox issues)

Of the 5 I mentioned that had I suspected of having CI issues, `dnsprobe` and `augustus` are now working.

For the issues where I suspect the blame falls on the CI infrastructure, where would be the best place to open an issue?  As a `homebrew-core`?  Or a `brew` issue?  Elsewhere?   There does seem to be some underlying sandbox issue where formula whose tests use some combination of networking and pty pass on 10.14/10.15/11-arm but fail on 11-intel.

**Comment by mitchblank** [comment]
> 0.14/10.15/11-arm but fail on 11-intel

Update: based on the conversation at https://github.com/Homebrew/homebrew-core/pull/71665#issuecomment-782916530 I am no longer sure how many have had 11-arm actually succeed and how many were merely skipped on 11-arm because of the lack of a preexisting arm bottle

**Comment by mitchblank** [comment]
> ❌ @dtrodrigues bottle request for boost-python [failed](https://github.com/Homebrew/homebrew-core/actions/runs/587742869).

boost-python compiled OK on ARM, but `brew test` failed due to stricter `dlopen()` requirements I guess?
```
==> python
Traceback (most recent call last):

  File "<stdin>", line 2, in <module>
ImportError: dlopen(./hello.so, 2): no suitable image found.  Did find:
	file system relative paths not allowed in hardened programs
An exception occurred within a child process:
```

**Comment by mitchblank** [comment]
> ❌ @dtrodrigues bottle request for cvs [failed](https://github.com/Homebrew/homebrew-core/actions/runs/587761343).

Failed due to autoconf being too old I guess:
```
checking build system type... configure: error: /bin/sh build-aux/config.sub -apple-darwin20.3.0 failed
```
Normally this is easy to fix just by rerunning autotools.  A bit more complicated in this case since we have existing patches that mess with `configure` which would have to be rethought :-/

**Comment by mitchblank** [comment]
> ❌ @dtrodrigues bottle request for gnu-prolog [failed](https://github.com/Homebrew/homebrew-core/actions/runs/587762703).
```
configure: error: unsupported architecture arm-apple-darwin20.3.0
```
Not clear if it's just an autoconf-too-old issue there or if the package itself doesn't support ARM currently.

**Comment by mitchblank** [comment]
> ❌ @dtrodrigues bottle request for hdf5@1.8 [failed](https://github.com/Homebrew/homebrew-core/actions/runs/587763390).

Looks like we have a sha256 mismatch to track down there...
```
==> Verifying checksum for '238c78407b5ce3d60979d2814cf24d6a76a0c7423c600ce4ce692d25915ec8d9--hdf5-1.8.22.tar.bz2'
Error: SHA256 mismatch
Expected: 0ac77e1c22bce5bbbdb337bd7f97aeb5ef43c727a84ccb6d683d092eb57ebd8e
  Actual: 689b88c6a5577b05d603541ce900545779c96d62b6f83d3f23f46559b48893a4
```

**Comment by mitchblank** [comment]
> ❌ @dtrodrigues bottle request for neko [failed](https://github.com/Homebrew/homebrew-core/actions/runs/587764134).

This looks like actually the `bdw-gc` dependency is built incorrectly in the ARM bottle!
```
ld: warning: ignoring file /opt/homebrew/lib/libgc.dylib, building for macOS-x86_64 but attempting to link with file built for macOS-arm64
```

Maybe something that `brew audit` should look for?  This would also break a few other formulae:
```
$ grep -l bdw-gc *.rb
crystal.rb
ecl.rb
guile.rb
guile@2.rb
kite.rb
neko.rb
ngs.rb
sagittarius-scheme.rb
w3m.rb
zile.rb
```

**Comment by mitchblank** [comment]
> ❌ @dtrodrigues bottle request for odin [failed](https://github.com/Homebrew/homebrew-core/actions/runs/587764458).

Looks like this one is fixed upstream: https://github.com/odin-lang/Odin/commit/776c3f4e90f5a8c2240ea084ffe94cce9f5525aa#diff-1f0034637c3196a3296a2a9673068256f899131064f94daad4623dd1506d5b90 

But not clear if it's fully done yet since there is still an open PR for arm support: https://github.com/odin-lang/Odin/pull/738

**Comment by mitchblank** [comment]
> ❌ @dtrodrigues bottle request for pg_top [failed](https://github.com/Homebrew/homebrew-core/actions/runs/587764548).

Looks like we need to add a re-run of autoconf here:
```
checking build system type... configure: error: /bin/sh ./config.sub -apple-darwin20.3.0 failed
```

**Comment by mitchblank** [comment]
> ❌ @dtrodrigues bottle request for po4a [failed](https://github.com/Homebrew/homebrew-core/actions/runs/587764946).

Something failed to build in some perl module on ARM, unfortunately not clear what from the CI output:
```
==> /opt/homebrew/Cellar/po4a/0.62/bin/po4a-gettextize -f asciidoc -m en.tex -p out.pot
dyld: missing lazy symbol called
```

**Comment by mitchblank** [comment]
> ❌ @dtrodrigues bottle request for speech-tools [failed](https://github.com/Homebrew/homebrew-core/actions/runs/587768379).

Looks like upstream doesn't support arm64 yet?
```
==> make
Check system type
config/config:200: config/systems/arm64_Darwin.mak: No such file or directory
../config/config:200: ../config/systems/arm64_Darwin.mak: No such file or directory
```
That file doesn't exist in the upstream git yet, either.

**Comment by mitchblank** [comment]
> ❌ @dtrodrigues bottle request for spidermonkey [failed](https://github.com/Homebrew/homebrew-core/actions/runs/587768541).

Looks like some assumptions about compiler flags are baked into its configure script:
```
checking for valid optimization flags... no
configure: error: These compiler flags are invalid: -O3 -fomit-frame-pointer
```
Note that we build against a pretty old version of spidermonkey for dependency purposes so patching will probably be required to make this ARM buildable, if that is practical at all.

**Comment by mitchblank** [comment]
> ❌ @dtrodrigues bottle request for rethinkdb [failed](https://github.com/Homebrew/homebrew-core/actions/runs/587765281).

Looks like their patched version of v8 is too old to build on OS/X-plus-ARM:
```
./google/protobuf/stubs/atomicops_internals_macosx.h:162:50: error: unknown type name 'Atomic64'; did you mean 'Atomic32'?
inline Atomic64 Barrier_AtomicIncrement(volatile Atomic64* ptr,
                                                 ^~~~~~~~
                                                 Atomic32
./google/protobuf/stubs/atomicops.h:65:15: note: 'Atomic32' declared here
typedef int32 Atomic32;
```
Seems to be something under active development so hopefully there will be an upstream release addressing it sooner or later.

**Comment by mitchblank** [comment]
> ❌ @dtrodrigues bottle request for tarantool [failed](https://github.com/Homebrew/homebrew-core/actions/runs/587768732).
```
<inline asm>:30:2: error: unknown directive
        .type coro_startup, %function
        ^
1 error generated.
make[2]: *** [CMakeFiles/coro.dir/third_party/coro/coro.c.o] Error 1
```
Seems to possibly be a known issue with some bundled 3rd party code that is conditionally enabled? https://github.com/tarantool/tarantool/pull/5684

**Comment by mitchblank** [comment]
> ❌ @dtrodrigues bottle request for zpaq [failed](https://github.com/Homebrew/homebrew-core/actions/runs/587769984).

This one is weird, on ARM it built OK but the unit test failed with `allocx: failed`:
```
./zpaq extract archive.zpaq zpaq -to zpaq.new
zpaq v7.15 journaling archiver, compiled Feb 21 2021
archive.zpaq: Skipping block at 181396: allocx failed
1 versions, 0 files, 6 fragments, 0.181396 MB
Extracting 0.000000 MB in 0 files -threads 8
0.001 seconds (all OK)
cmp zpaq zpaq.new
cmp: zpaq.new: No such file or directory
make: *** [check] Error 2
```

One guess is possibly we need to compile with `NOJIT` on ARM?   Looking at https://github.com/zpaq/zpaq/blob/master/libzpaq.h
```
  -DNOJIT   Don't assume x86-32 or x86-64 with SSE2 (slower).
```

**Comment by mitchblank** [comment]
> ❌ @dtrodrigues bottle request for jack [failed](https://github.com/Homebrew/homebrew-core/actions/runs/587786253).

Looks like it's more CPU specific than I would have guessed:
```
../common/JackAtomic.h:31:15: error: use of undeclared identifier 'CAS'
    } while (!CAS(actual, actual + 1, val));
              ^
```
We are a version behind on that formula and @fxcoudert has already looked into it (but the PR went stale) #69123

**Comment by mitchblank** [comment]
> ❌ @dtrodrigues bottle request for ponyc [failed](https://github.com/Homebrew/homebrew-core/actions/runs/587765142).

```
/tmp/ponyc-20210221-3290-aquuat/build/libs/bin/llc: error: : error: unable to get target for 'arm-apple-darwin20.3.0', see --version and --triple.
make[3]: *** [except_try_catch.o] Error 1
```
Looks like just lack of upstream support.

**Comment by mitchblank** [comment]
> * `nauty` failed to build because the website is down.  I emailed the project maintainer who is now looking into it.

@fxcoudert -- could you try another 11-Intel rebottle on `nauty`?  The previous failure was due to a power outage that took out the upstream's homepage/download   That at least should be fixed now.

**Pull request closed without merging**